provider/pkg/await/await.go

```diff
@@ -834,6 +835,27 @@ func Deletion(c DeleteConfig) error {
 		PropagationPolicy: &deletePolicy,
 	}
 
+	live, err := client.Get(c.Context, c.Name, metav1.GetOptions{})
+	if err != nil {
+		return nilIfGVKDeleted(err)
+	}
+
+	actualSSAManagers := sets.Set[string]{}
+	for _, f := range live.GetManagedFields() {
+		// Ignore fields not managed by pulumi SSA.
+		if !strings.HasPrefix(f.Manager, "pulumi-kubernetes-") {
+			continue
+		}
+		actualSSAManagers = actualSSAManagers.Insert(f.Manager)
+	}
+	if c.ServerSideApply && !actualSSAManagers.Has(c.FieldManager) {
```

Comment:
As written this would also no-op in cases where the object was updated with CSA. 

We can narrow this by also checking `actualSSAManagers` is non-empty, so we know the live object was at least touched by pulumi.